gdc_client/parcel/client.py

```diff
@@ -216,19 +199,30 @@ def _download(self, nprocs, stream):
         n_procs = 1 if stream.size < 0.01 * const.GB else nprocs
         producer = SegmentProducer(stream, n_procs)
 
+        if producer.done:
+            return
+
         def download_worker():
             while True:
                 try:
                     segment = producer.q_work.get()
                     if segment is None:
                         return log.debug("Producer returned with no more work")
                     stream.write_segment(segment, producer.q_complete)
+                    # write_segment completed successfully, send sentinal value
```

Comment:
```suggestion
                    # write_segment completed successfully, send sentinel value
```
?

and can you update L210 to not `return log.debug`. a break statement after the log should do the job

Reply:
i can't spell, so i'll change the spelling of sentinel everywhere. 

so L210, it does need to return, as the worker process needs to exit when the main process sends None as a sentinel value to tell it to exit. I can move the debug line above it though and just have a separate return statement under it